FullStorySegmentMiddleware/Classes/FullStoryMiddleware.m

```diff
@@ -124,6 +128,96 @@ - (SEGPayload *) getNewPayloadWithFSURL:(SEGContext * _Nonnull)context {
     return newPayload;
 }
 
+
+- (NSDictionary *) getSuffixedProps: (NSDictionary *)properties{
+    //TODO: Segment will crash and not allow props to have curcular dependency, but we should handle it here anyways
+    
+    NSMutableDictionary *props = [[NSMutableDictionary alloc] initWithCapacity:[properties count]];
+    for(id key in properties){
+        NSString *suffix = @"";
+
+        // properties should always be a NSDictionary, but still check to make sure the value is an object
+        // Check Type Encoding: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ObjCRuntimeGuide/Articles/ocrtTypeEncodings.html
+        const char* typeCode = @encode(typeof([props valueForKey:key]));
+
+        if([@"@" isEqualToString:@(typeCode)]){
+            NSObject *obj = [properties valueForKey:key];
+            
+            // if it's a NSDictionary: recurrsively get props, and we can not take underscore in key for nested objects, so we just remove them
+            // more info: https://help.fullstory.com/hc/en-us/articles/360020623234-FS-Recording-Client-API-Requirements
+            if([obj isKindOfClass:[NSDictionary class]]){
+                [props setValue:[self getSuffixedProps:[properties valueForKey:key]]
+                         forKey:[key stringByReplacingOccurrencesOfString:@"_" withString:@""]];
```

Comment:
Hmm, I actually mind this a little less.  I don't think it's on a hot path, and burning a little extra memory to be a little clearer about what's going on is an OK tradeoff.  (That said, it ought be a conscious tradeoff!)

Reply:
Thanks for the inputs! I didn't think about it but now I went through the documents on what's likely being passed in here, it should not have a lot of input cases where it will enter `[obj isKindOfClass:[NSDictionary class]]`, even thought the key is not that likely to have underscore in this case, but it should happen every once in a while. so I think this would be fine.